nibe/connection/encoders.py

```diff
@@ -34,71 +35,141 @@
     }
 )
 
+integer_limit = {
+    "u8": 0xFF,
+    "s8": -0x80,
+    "u16": 0xFFFF,
+    "s16": -0x8000,
+    "u32": 0xFFFFFFFF,
+    "s32": -0x80000000,
+}
+
+
+_RawDataT = TypeVar("_RawDataT")
 
-class CoilDataEncoder:
+
+class CoilDataEncoder(Generic[_RawDataT]):
     """Encode and decode coil data."""
 
     word_swap: Optional[bool] = None
 
     def __init__(self, word_swap: Optional[bool] = None):
         self.word_swap = word_swap
 
-    def encode(self, coil_data: CoilData) -> bytes:
+    @abstractmethod
+    def encode_raw_value(self, size: str, raw_value: int) -> _RawDataT:
+        pass
+
+    def encode(self, coil_data: CoilData) -> _RawDataT:
         """Encode coil data to bytes.
 
         :raises EncodeException: If encoding fails"""
         try:
             coil_data.validate()
```

Comment:
Validate will throw exception if coil_data value is None

Reply:
True. I wonder if we should allow it and set limit value instead.

Reply:
Also.. i wonder if encode/decode should move into CoilData structure now instead?

Reply:
I do not want to let library users to send undefined values to the pump. I have no idea what consequences it can bring.
CoilData is quite pure, I think we should not complicate it.

Reply:
I am even thinking the opposite, if we want to move all raw related code to Coder instead. From Coil and CoilData.